RecoEcal/EgammaClusterAlgos/src/PFECALSuperClusterAlgo.cc

```diff
@@ -514,6 +538,15 @@ void PFECALSuperClusterAlgo::finalizeSuperCluster(CalibratedPFCluster& seed,
   new_sc.setEtaWidth(pfwidth.pflowEtaWidth());
   new_sc.setPhiWidth(pfwidth.pflowPhiWidth());
 
+  float isol = 0;
+  edm::Handle<reco::TrackCollection> trackHandle;
+  iEvent.getByToken(trackProducer_, trackHandle);
+  if (trackHandle.isValid()) {
+    const reco::TrackCollection* trackCollection = trackHandle.product();
+    isol = isoCalculator_->scPtSum(new_sc, trackCollection);
+  }
+  new_sc.setTrackIsolation(isol);
```

Comment:
it may be better to just not set the isolation if there are no tracks (ie move it all in the if statement) as then it would be the default initial value of the supercluster rather than there being a default value if its not a mustache and a seperate but currently identical default value if its mustache but has no tracks in the event